crates/workspace/src/workspace.rs

```diff
@@ -3122,21 +3123,46 @@ impl Workspace {
         }))
     }
 
+    pub fn join_all_panes(&mut self, cx: &mut ViewContext<Self>) {
+        let active_item = self.active_pane.read(cx).active_item().take();
+        for pane in self.panes.clone() {
+            self.move_all_items(cx, pane.clone(), self.active_pane.clone().clone());
+        }
+        if let Some(active_item) = active_item {
+            self.activate_item(&(*active_item), true, true, cx);
+        }
+        cx.notify();
+    }
+
     pub fn join_pane_into_next(&mut self, pane: View<Pane>, cx: &mut ViewContext<Self>) {
-        let next_pane = self
-            .find_pane_in_direction(SplitDirection::Right, cx)
-            .or_else(|| self.find_pane_in_direction(SplitDirection::Down, cx))
-            .or_else(|| self.find_pane_in_direction(SplitDirection::Left, cx))
-            .or_else(|| self.find_pane_in_direction(SplitDirection::Up, cx));
-        let Some(next_pane) = next_pane else {
+        let Some(next_pane) = self.find_next_pane(cx) else {
             return;
         };
+        self.move_all_items(cx, pane, next_pane);
+        cx.notify();
+    }
 
-        let item_ids: Vec<EntityId> = pane.read(cx).items().map(|item| item.item_id()).collect();
+    fn move_all_items(
+        &mut self,
+        cx: &mut ViewContext<'_, Workspace>,
```

Comment:
Please move `cx` to the last place in the argument list, as it's done in 99% of the code.

```suggestion
        cx: &mut ViewContext<'_, Self>,
```